src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayController.cs

```diff
@@ -17,46 +20,75 @@ namespace SFA.DAS.ApplyService.InternalApi.Controllers
     public class RoatpGatewayController: Controller
     {
         private readonly IApplyRepository _applyRepository;
+        private readonly IInternalQnaApiClient _qnaApiClient;
+        private readonly IGatewayApiChecksService _gatewayApiChecksService;
         private readonly ILogger<RoatpGatewayController> _logger;
 
         /// <summary>
         /// 
         /// </summary>
         /// <param name="applyRepository"></param>
-        public RoatpGatewayController(IApplyRepository applyRepository, ILogger<RoatpGatewayController> logger)
+        public RoatpGatewayController(IApplyRepository applyRepository, ILogger<RoatpGatewayController> logger, IInternalQnaApiClient qnaApiClient, IGatewayApiChecksService gatewayApiChecksService) 
         {
             _applyRepository = applyRepository;
             _logger = logger;
+            _qnaApiClient = qnaApiClient;
+            _gatewayApiChecksService = gatewayApiChecksService;
         }
 
         [Route("Gateway/Page/Submit")]
         [HttpPost]
          public async Task GatewayPageSubmit([FromBody] UpsertGatewayPageAnswerRequest request)
         {
-            _logger.LogInformation($"RoatpGatewayController-GatewayPageSubmit - ApplicationId '{request.ApplicationId}' - PageId '{request.PageId}' - Status '{request.Status}' - UserName '{request.UserName}' - PageData '{request.GatewayPageData}'");
-            try
-            {
-                await _applyRepository.SubmitGatewayPageAnswer(request.ApplicationId, request.PageId, request.Status, request.UserName, request.GatewayPageData);
-            }
-            catch(Exception ex)
-            {
-                _logger.LogError(ex, "RoatpGatewayController-GatewayPageSubmit - Error: '" + ex.Message + "'");
-            }
-            
+            await _applyRepository.SubmitGatewayPageAnswer(request.ApplicationId, request.PageId, request.UserName,
```

Comment:
I don't usually agree with controllers (or the UI/API layer at all) although in this instance it's probably simple enough to live with it.